build/toolchain/linux_arm/linux_arm_toolchain.gni

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+
+import("${build_root}/config/android/config.gni")
+import("${build_root}/toolchain/gcc_toolchain.gni")
+
+declare_args() {
+  # similar to "CROSS_COMPILE" when build kernel
+  linux_arm_cross_compile = ""
+}
+
+template("linux_arm_toolchain") {
+  if (linux_arm_cross_compile == "") {
+    assert(
+        false,
+        "assign a value to the build argument 'linux_arm_cross_compile' when use linux_arm_toolchain")
+  }
+
+  _invoker_toolchain_args = invoker.toolchain_args
+  _linux_arm_toolchain_args = {
+    current_os = invoker.current_os
+    is_clang = false
+    forward_variables_from(_invoker_toolchain_args, "*")
+  }
+
+  gcc_toolchain(target_name) {
+    toolchain_args = _linux_arm_toolchain_args
+    ar = linux_arm_cross_compile + "ar"
```

Comment:
Hi @mspang 
it's a good idea to make separate arguments for each tool.

For you second comment, in fact, I used to specify the toolchain with build argument "custom_toolchain", and then found it a bit unease of use:
1. when I use gn at the top direcory of connectedhomeip, I specify the custom toolchain like: custom_toolchain="//build/toolchain/linux_arm:linux_arm"
2. when I use gn at a subdirectory like "examples/lighting-app/linux/", I specify the custom toolchain like: custom_toolchain="//third_party/connectedhomeip/build/toolchain/linux_arm:linux_arm"

Different is used to specify the same toolchain, For me this is totally okay, but I think that this may be a bit confusing for customers who are beginners of gn tools. Can I made the change to "build/config/BUILDCONFIG.gn" like below?
`
else if (target_os == "linux" &&
           (target_cpu == "arm64" || target_cpu == "arm") &&
           target_cpu != host_cpu) {
  _default_toolchain = "${_build_overrides.build_root}/toolchain/linux_arm"
}
`

Reply:
> Hi @mspang
> it's a good idea to make separate arguments for each tool.
> 
> For your second comment, in fact, I used to specify the toolchain with build argument "custom_toolchain", and then found it a bit unease of use:
> 
> 1. when I use gn at the top direcory of connectedhomeip, I specify the custom toolchain like: custom_toolchain="//build/toolchain/linux_arm:linux_arm"
> 2. when I use gn at a subdirectory like "examples/lighting-app/linux/", I specify the custom toolchain like: custom_toolchain="//third_party/connectedhomeip/build/toolchain/linux_arm:linux_arm"
> 
> Different string is used to specify the same toolchain, For me this is totally okay, but I think that this may be a bit confusing for customers who are beginners of gn tools.

What about

```
gn gen out/custom --args="
    import(\"//build_overrides/build.gni\")
    custom_toolchain=\"${build_root}/toolchain/linux_arm:linux_arm\"
    target_cc = \"${CROSS_COMPILE}-gcc\"
    target_cxx = \"${CROSS_COMPILE}-g++\"
    target_cxx = \"${CROSS_COMPILE}-ar\"
"
```

Otherwise I think defining a new argument and doing something like

```
gn gen out/custom --args="
    use_custom_target_toolchain = true
    target_cc = \"${CROSS_COMPILE}-gcc\"
    target_cxx = \"${CROSS_COMPILE}-g++\"
    target_cxx = \"${CROSS_COMPILE}-ar\"
"
```

would be fine.

> Can I made the change to "build/config/BUILDCONFIG.gn" like below?
> `else if (target_os == "linux" && (target_cpu == "arm64" || target_cpu == "arm") && target_cpu != host_cpu) { _default_toolchain = "${_build_overrides.build_root}/toolchain/linux_arm" }`

I don't think we should couple providing a toolchain to cross compiling on Linux. Linux ARM[64] is a very common target, and we already need to support it well for Raspberry Pi. This should be made to *work* by using the compiler that running the "boostrap" process provides you:

`gn gen out/arm --args='target_cpu="arm" is_clang=true'`

Reply:
Thank you, @mspang , I've updated the PR based on your comments.